### PR TITLE
Added missing docstring for inner product.

### PR DESCRIPTION
**Pull request by MichaelBroughton**
Accidentally merged without adding. Here is the docstring snippets.

### CODE REVIEW

File: tensorflow_quantum/core/ops/math_ops/inner_product_op.py [hunk 1]
```diff
@@ -23,12 +23,45 @@
 def inner_product(programs, symbol_names, symbol_values, other_programs):
     """Calculate the inner product between circuits.
 
+    Compute (potentailly many) inner products betwen the given circuits and
```
Comment (zaqqwerty) [inline tensorflow_quantum/core/ops/math_ops/inner_product_op.py:26]: spelling, `potentailly` ==> `potentially`

File: tensorflow_quantum/core/ops/math_ops/inner_product_op.py [hunk 2]
```diff
@@ -23,12 +23,45 @@
...
+    the symbol free comparison circuits.
+
     Calculates out[i][j] = \langle \psi_{\text{programs[i]}} \\
         (\text{symvol_values[i]}) | \psi_{\text{other_programs[j]}} \rangle
```
Comment (zaqqwerty) [inline tensorflow_quantum/core/ops/math_ops/inner_product_op.py:30]: spelling, `symvol_values` ==> `symbol_values`

**Review by zaqqwerty — CHANGES_REQUESTED** [review]
spelling nits

File: tensorflow_quantum/core/ops/math_ops/inner_product_op.py (hunk 1 above)
Comment (MichaelBroughton) [reply tensorflow_quantum/core/ops/math_ops/inner_product_op.py]: Fixed.

File: tensorflow_quantum/core/ops/math_ops/inner_product_op.py (hunk 2 above)
Comment (MichaelBroughton) [reply tensorflow_quantum/core/ops/math_ops/inner_product_op.py]: Fixed.

**Review by zaqqwerty — APPROVED** [review]
LGTM

**Pull request merged**